### PR TITLE
Fix broken KeySupport for Gameshell (and others)

### PR DESCRIPTION
**Pull request by danie1kr**
Hi all,
I noticed that a recent change to controller.cpp broke the Gameshell support as `ToControllerButtonEvent` no longer reacts to key presses in the menu and later. With this commit, it is fixed.
Looking at the history, this change might have been forgotten when df56d6a03bdf1f148beafa8cbc8612eac4f6c34b was done.
Hope this fixes the issue also for other devices which need the `HAS_KBCTRL == 1` define.
Happy coding!

### CODE REVIEW

**Comment by AJenbo** [comment]
@danie1kr looks like this breaks compiling :(

File: SourceX/controls/controller.cpp [hunk 1]
```diff
@@ -22,7 +22,11 @@ ControllerButtonEvent ToControllerButtonEvent(const SDL_Event &event)
 	default:
 		break;
 	}
-
+#if HAS_KBCTRL == 1
+	result.button = KbCtrlToControllerButton(event);
+	if (result.button != ControllerButton_NONE)
+		return result;
+#endifs
```
Comment (yuripourre) [inline SourceX/controls/controller.cpp:29]: #endif

**Comment by yuripourre** [comment]
@danie1kr Nice catch! Your code has a minor typo.

File: SourceX/controls/controller.cpp (hunk 1 above)
Comment (AJenbo) [reply SourceX/controls/controller.cpp:29]: image

Comment (AJenbo) [reply SourceX/controls/controller.cpp:29]:
```suggestion
#endif
```

**Review by glebm — APPROVED** [review]

**Comment by AJenbo** [comment]
@danie1kr thanks for fixing this.

**Comment by danie1kr** [comment]
> @danie1kr thanks for fixing this.

sure, happy to help.
Sorry for the git-typo, it was late...
But it works on my device.

**Pull request merged**